addOns/reports/src/main/java/org/zaproxy/addon/reports/sarif/SarifBase64Encoder.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.reports.sarif;
+
+import java.util.Base64;
+
+public class SarifBase64Encoder {
+
+    public static final SarifBase64Encoder DEFAULT = new SarifBase64Encoder();
+
+    public String encodeBytesToBase64(byte[] bytes) {
+        if (bytes == null) {
+            return null;
+        }
+        byte[] encoded = Base64.getEncoder().encode(bytes);
+        return new String(encoded);
+    }
+}
```

Comment:
A util class is normally full of static methods and contains no non static ones.

I do not like usage of static methods and there is one main reason: 
When using objects instead of static utility methods it's much more easy to make atomic tests. The 
`SarifResultTest` does mock the `SarifBase64Encoder` and also the `SarifBinaryContentDetector` and inject those into tested class instance (SarifResult) instead of the default one.

So the SarifResult test does not check if base 64 encoding is done correctly but only mocks some simple base64 psuedo encoding and also simulates the binary content detection - enough for testing. 
When somebody uses static methods and wants to have also the test separation as mentioned before, he/she normally comes up with tools like PowerMock etc. Which - IMHO - is a bad taste.

Also I prefer a class having an explicit purpose which is clear when reading the class name. This class is just a base64 encoder - not more not less. And it becomes clear because of the naming and the provided methods.
So tests will concentrate only on the purpose of this class - in this case only the base 64 encoding (simplification). Same for `SarifBinaryContentDetector`.

Util class tests do test multiple things (each provided method) - so those tests normally also get a little bit messy/full when the util classes are growing.

Reply:
Okay, I'm good with that explanation/thought process.